tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -88,41 +89,109 @@ class AppFilterImpl(
+            val keyValueConfigs = individualConfigs.filter(kvConfig => kvConfig.size == 2)
+            val keysOnlyConfigs = individualConfigs.filter(kconfig => kconfig.size != 2)
+
+            if (keyValueConfigs.nonEmpty || keysOnlyConfigs.nonEmpty) {
+              val validConfigsMap = keyValueConfigs.map(a => a(0) -> a(1)).toMap
+              val allConfigs = sparkProp.configName // all configs from eventlog
+              val allConfigKeys = sparkProp.configName.keys.toList // for keys only confs
+              //Intersection of configs provided in the filter args with event log configs.
+              val commonConfigs = validConfigsMap.keySet.intersect(allConfigs.keySet).map(
+                k => (k, allConfigs(k))).toMap
+
+              validConfigsMap.filter { configArgs =>
+                commonConfigs.exists(eventLogConf =>
+                  configArgs._1 == eventLogConf._1 && configArgs._2 == eventLogConf._2)
+              }.nonEmpty || sparkProperties.intersect(allConfigKeys).nonEmpty
+            } else {
+              false
+            }
+          }
+        }
+        configFilteredResult
+      } else { // include all eventlogs whose keys match
+        val configFilteredResult = userNameLogicFiltered.filter { appFilterReturnParameters =>
+          appFilterReturnParameters.appInfo.sparkProperties.exists { sparkProp =>
+            val allConfigsKeys = sparkProp.configName.keys.toList
+            sparkProperties.intersect(allConfigsKeys).nonEmpty
+          }
+        }
+        configFilteredResult
+      }
+    } else {
+      userNameFiltered
+    }
+
     val appTimeFiltered = if (appArgs.startAppTime.isSupplied) {
       val msTimeToFilter = AppFilterImpl.parseAppTimePeriodArgs(appArgs)
       val logicFiltered = if (appArgs.any()) {
         apps
       } else {
-        userNameFiltered
+        configFiltered
       }
-      logicFiltered.filter(_.appInfo.exists(_.startTime >= msTimeToFilter))
+      logicFiltered.filter(_.appInfo.appStartInfo.exists(_.startTime >= msTimeToFilter))
+    } else {
+      configFiltered
+    }
+
+    // If `any` is specified as logicFilter config, then store the value in variable which is used
+    // later for final computation. These are computed separately for all filters.
+    val appNameFilteredForDisjunction = if (filterAppName.nonEmpty && appArgs.any()) {
+      appNameFiltered
     } else {
+      Seq.empty[AppFilterReturnParameters]
+    }
+
+    val userNameFilteredForDisjunction = if (userName.nonEmpty && appArgs.any()) {
       userNameFiltered
+    } else {
+      Seq.empty[AppFilterReturnParameters]
+    }
+
+    val configFilteredForDisjunction = if (sparkProperties.nonEmpty && appArgs.any()) {
+      configFiltered
+    } else {
+      Seq.empty[AppFilterReturnParameters]
+    }
+
+    val appTimeFilteredForDisjunction = if (appArgs.startAppTime.isSupplied && appArgs.any()) {
+      appTimeFiltered
+    } else {
+      Seq.empty[AppFilterReturnParameters]
```

Comment:
There was a bug here - for disjunction(any) with only one filter parameter,  if the filter did not match to any eventlogs, then all the event logs were selected. Reason being `appNameFiltered` would be set to `apps` and then `userNameFiltered` would be to `appNameFiltered`. So in the code below finalLogicFiltered would be set to `apps` which is wrong.  So now I am making sure finalLogicFiltered gets the correct value. Added a test to veridy this.